azkaban-cli/src/main/groovy/com/linkedin/gradle/util/HttpUtil.groovy

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.linkedin.gradle.util;
+
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.util.EntityUtils;
+
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+
+/**
+ * Utility class for handling HTTP request and responses.
+ * <p>
+ * This is currently used by azkaban classes for HTTP request handling.
+ */
+public class HttpUtil {
+
+  private final static Logger logger = Logging.getLogger(this.class);
+
+  /**
+   * batchGet requests implementation using threads
+   *
+   * @param uriList List of uri's for performing GET requests
+   * @return responseList The list of all the batch responses
+   */
+  static List<String> batchGet(List<URI> uriList) {
+    PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+    connectionManager.setDefaultMaxPerRoute(20);
+    CloseableHttpClient httpClient = HttpClients.custom().setConnectionManager(connectionManager).build();
+
+    // create a thread for each URI
+    GetThread[] threads = new GetThread[uriList.size()];
```

Comment:
I have moved return to try block as finally always executes unless a system call for exit is triggered or if jvm crashes.